src/librustc/middle/resolve_lifetime.rs

```diff
@@ -1673,6 +1673,12 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                         }
                         _ => None,
                     } {
+                        if name.name.as_str() == "'_" {
```

Comment:
If you're only going to check for `'_` :
```suggestion
                        if name.name == kw::UnderscoreLifetime {
```